nest/nest.py

```diff
@@ -228,6 +228,9 @@ def _repr_name(self):
 
         return self.where
 
+    def __str__(self):
+        return str(self._device)
+
```

Comment:
I think this should be the `__repr__` not the `__str__`.  Currently doing `print('%s' % napi.thermostats[0])` results in `<Thermostat: Downstairs>`. This would make it dump the underlying dict, which while nice for development is not so nice for usability.

Currently all the formating is done by the existing `__repr__`s,  so the just flipping them to `__str__` and these to `__repr__`:

```diff
diff --git a/nest/nest.py b/nest/nest.py
index c383950..9c11a5d 100644
--- a/nest/nest.py
+++ b/nest/nest.py
@@ -182,7 +182,7 @@ class NestBase(object):
         self._serial = serial
         self._nest_api = nest_api

-    def __repr__(self):
+    def __str__(self):
         return '<%s: %s>' % (self.__class__.__name__, self._repr_name)

     def _set(self, what, data):
@@ -228,7 +228,7 @@ class Device(NestBase):

         return self.where

-    def __str__(self):
+    def __repr__(self):
         return str(self._device)

     @property
@@ -991,7 +991,7 @@ class CameraEvent(NestBase):
     def _event(self):
         return self._camera.get('last_event')

-    def __repr__(self):
+    def __str__(self):
         return '<%s>' % (self.__class__.__name__)

     def activity_in_zone(self, zone_id):
@@ -1332,7 +1332,7 @@ class Structure(NestBase):
     def _structure(self):
         return self._nest_api._status[STRUCTURES][self._serial]

-    def __str__(self):
+    def __repr__(self):
         return str(self._structure)

     def _set_away(self, value, auto_away=False):
```

Then `print` still looks nice:

```python
>>> print(napi.thermostats[0])
<Thermostat: Downstairs> 
```

And the repr is there to make dev easier:
```python
>>> print(repr(napi.thermostats[0]))
{'humidity': 50,  ...
```